### PR TITLE
Initial work on inheritance

### PR DESCRIPTION
**Pull request by anodos325**
nfs4-acl-tools originally lacked important functionality to provide
usable management tools at scale. This commit adds two new functions

acl_nfs4_strip()
acl_is_trivial_np()

Use-case is similar to acl_strip_np(3) and acl_is_trivial_np(3) in
FreeBSD.

The former returns allocates and returns a trivial ACL (one that can
be expressed via POSIX permissions without loss of information). The
latter is to be used to check whether an NFSv4 ACL is trivial.

Accordingly, nfsxdr_setfacl is given a new flag `-b` which can be used
to strip the ACL from a file, and nfs4xdr_getfacl output has been
expanded to include a comment indicating whether the ACL is trivial.

### CODE REVIEW

**Comment by anodos325** [comment]
Regression tests will be handled through middleware API tests for NFSv4 ACLs.

File: libnfs4acl/acl_nfs4_inheritance.c [hunk 1]
```diff
@@ -0,0 +1,563 @@
...
+bool acl_nfs4_sync_mode_from_acl(mode_t *_mode, struct nfs4_acl *aclp)
+{
+	mode_t old_mode = *_mode, mode = 0, deny_mode = 0;
+	struct nfs4_ace *ace = NULL;
+
+	for (ace = nfs4_get_first_ace(aclp); ace != NULL;
+	     ace = nfs4_get_next_ace(&ace)) {
+		if (ace->type != NFS4_ACE_ACCESS_ALLOWED_ACE_TYPE &&
+		    ace->type != NFS4_ACE_ACCESS_DENIED_ACE_TYPE) {
+			fprintf(stderr, "Invalid ACE type: %d\n", ace->type);
```
Comment (yocalebo) [inline libnfs4acl/acl_nfs4_inheritance.c:372]: Is this a common scenario? If so, could this barf to stderr quite a bit? Maybe you want that to happen? Maybe hide it behind the `NFS4_DEBUG` compile time option?  Just curious mostly.

Comment (anodos325) [reply libnfs4acl/acl_nfs4_inheritance.c:372]: This would be quite unexpected since we don't allow setting audit / alarm ACEs.

Comment (yocalebo) [reply libnfs4acl/acl_nfs4_inheritance.c:372]: Cool, mostly inquisitive about that so appreciate the explanation.

**Review by yocalebo — APPROVED** [review]

**Pull request merged**